### PR TITLE
GenZ Sprint Two 39: Solution added for the extension file organizer

### PR DESCRIPTION
**Pull request by sarah-alshukaily**

### CODE REVIEW

**Review by Rudainasaleh — APPROVED** [review]

**Review by aljab017 — APPROVED** [review]

**Review by Sughiya-AlSaid — DISMISSED** [review]

File: examples/sara_work_area/genzsprtwo_39_files_oraganizer.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+miscellaneous_directory = "./dir/Miscellaneous"
+
+for directory in [image_directory, text_directory, miscellaneous_directory]:
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+try:
+    no_of_img_files, no_of_txt_files, no_of_misc_files = 0, 0, 0
+    files = os.listdir(parent_directory)
```
Comment (CodelineAtyab) [inline examples/sara_work_area/genzsprtwo_39_files_oraganizer.py:20]: We can make use of the built-in glob library to specifically get *.jpg, *.png etc list of file paths and then we can create dirs and move the relevant files to specific dirs.

File: examples/sara_work_area/genzsprtwo_39_files_oraganizer.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            no_of_img_files += 1
+        elif file_extension.lower() == ".txt":
+            destination_directory = text_directory
+            no_of_txt_files += 1
+        else:
+            destination_directory = miscellaneous_directory
+            no_of_misc_files += 1
+
+        new_file_path = os.path.join(destination_directory, file)
+        os.rename(os.path.join(parent_directory, file), new_file_path)
```
Comment (CodelineAtyab) [inline examples/sara_work_area/genzsprtwo_39_files_oraganizer.py:38]: We can make use of shitil module, which is specifically designed to handle file operations. 

**Definition**
shutil is a module in Python's standard library that provides a collection of high-level file operations. The name shutil stands for "shell utilities," and the module includes functions to help you copy, move, rename, and delete files and directories.

**Review by CodelineAtyab — CHANGES_REQUESTED** [review]
Please see my comments

File: examples/sara_work_area/genzsprtwo_39_files_oraganizer.py (hunk 1 above)
Comment (sarah-alshukaily) [reply examples/sara_work_area/genzsprtwo_39_files_oraganizer.py:20]: i prefer using os module in this case

File: examples/sara_work_area/genzsprtwo_39_files_oraganizer.py (hunk 2 above)
Comment (sarah-alshukaily) [reply examples/sara_work_area/genzsprtwo_39_files_oraganizer.py:38]: Changes made

**Review by aljab017 — APPROVED** [review]

**Review by Rudainasaleh — APPROVED** [review]

**Review by Alhasan-AJ — APPROVED** [review]

**Review by CodelineAtyab — APPROVED** [review]
Nice Work!

**Pull request merged**